packages/extension/language-client/index.ts

```diff
@@ -0,0 +1,183 @@
+import Notification from '@sqltools/core/contracts/notifications';
+import DepInstaller from './dep-installer';
+import { ErrorHandler } from '../api';
+
+export class SQLToolsLanguageClient {
+  public client: LanguageClient;
+  private clientErrorHandler: LanguageClientErrorHandler;
+  private avoidRestart: boolean;
+  private depInstaller: DepInstaller;
+
+  constructor(private context: ExtensionContext, public telemetry: Telemetry) {
+    this.client = new LanguageClient(
+      'SQLTools Language Server',
+      this.getServerOptions(),
+      this.getClientOptions(),
+      );
+    this.clientErrorHandler = this.client.createDefaultErrorHandler();
+    this.depInstaller = new DepInstaller(this, telemetry);
+
+    this.registerBaseNotifications();
+  }
+
+  public start(): Disposable {
+    return this.client.start();
+  }
+
+  public sendRequest<R, E, RO>(type: RequestType0<R, E, RO>, token?: CancellationToken): Thenable<R>;
+  public sendRequest<P, R, E, RO>(type: RequestType<P, R, E, RO>, params: P, token?: CancellationToken): Thenable<R>;
+  public sendRequest<R>(method: string, token?: CancellationToken): Thenable<R>;
+  public sendRequest<R>(method: string, param: any, token?: CancellationToken): Thenable<R>;
+  public async sendRequest() {
+    await this.client.onReady();
+    return this.client.sendRequest.apply(this.client, arguments);
+  }
+
+  public onRequest<R, E, RO>(type: RequestType0<R, E, RO>, handler: RequestHandler0<R, E>): void;
+  public onRequest<P, R, E, RO>(type: RequestType<P, R, E, RO>, handler: RequestHandler<P, R, E>): void;
+  public onRequest<R, E>(method: string, handler: GenericRequestHandler<R, E>): void;
+  public async onRequest() {
+    await this.client.onReady();
+    return this.client.onRequest.apply(this.client, arguments);
+  }
+
+  public sendNotification<RO>(type: NotificationType0<RO>): void;
+  public sendNotification<P, RO>(type: NotificationType<P, RO>, params?: P): void;
+  public sendNotification(method: string): void;
+  public sendNotification<T = any>(method: string, params: T): void;
+  public async sendNotification() {
+    await this.client.onReady();
+    return this.client.sendNotification.apply(this.client, arguments);
+  }
+  public onNotification<RO>(type: NotificationType0<RO>, handler: NotificationHandler0): void;
+  public onNotification<P, RO>(type: NotificationType<P, RO>, handler: NotificationHandler<P>): void;
+  public onNotification(method: string, handler: GenericNotificationHandler): void;
+  public async onNotification() {
+    await this.client.onReady();
+    return this.client.onNotification.apply(this.client, arguments);
+  }
+
+  private getServerOptions(): ServerOptions {
+    const serverModule = this.context.asAbsolutePath('languageserver.js');
+    const runOptions: NodeModule = {
+      module: serverModule,
+      transport: TransportKind.ipc,
+      runtime: process.platform !== 'win32' ? 'node' : 'node.cmd',
```

Comment:
Do you know if this is for Windows with WSL or all windows?
I've used as reference the bash-languageserver extension and it was using 'node.cmd' in the path. 

But since I haven't used WSL yet, I don't know.

Reply:
Node is as binary executable (node.exe) and is added to windows %PATH% to be executed globally. Not sure about WSL, but I think it is for all windows.

Reply:
But global `npm` packages are then created as .cmd scripts (e.g. npm.cmd, gulp.cmd, etc)...